src/lib.rs

```diff
@@ -179,6 +179,25 @@ impl<'a> SnakeCaseRef<'a> {
     }
 }
 
+#[cfg(feature = "const_literals")]
+/// an unsafe constructor for SnakeCaseRef. caller has to make sure the input is in fact valid.
+pub const unsafe fn from_str_unchecked(s: &str) -> SnakeCaseRef {
+    SnakeCaseRef(s)
+}
+#[cfg(feature = "const_literals")]
+/// this will construct a SnakeCafeRef with compile-time validation for string literals.
```

Comment:
```suggestion
/// this will construct a SnakeCafeRef with compile-time validation for string literals.
///
/// ```
/// use snake_case::scr_lit;
/// let snake_case = scr_lit!("my_little_snake");
/// ```
```

Reply:
can also add the failing case behind a comment ` // let bad_snake = scr_lit!("Python"); <-wont compile`